packages/docs/src/components/ExampleFooter.tsx

```diff
@@ -0,0 +1,95 @@
+/* eslint-disable react/no-danger */
+import React, { useRef, useState } from 'react';
+import uniqueId from 'lodash/uniqueId';
+import { ArrowIcon, Button } from '@cmsgov/design-system';
+import { highlightHtmlSyntax } from '../helpers/syntaxHighlighting';
+
+enum SnippetState {
+  CLOSED = 'closed',
+  HTML = 'html',
+  JSX = 'jsx',
+}
+
+interface ExampleFooterProps {
+  /**
+   * html string to prettify, highlight and display
+   */
+  html: string;
+  /**
+   * pre-highlighted jsx html as a string, because we gather it by scraping
+   */
+  highlightedJsx?: string;
+  /**
+   * A link or something to render to the right of the code button(s)
+   */
+  sourceLink?: React.ReactElement;
+}
+
+/**
+ * HTML code snippet. This component will prettify an html string and do syntax highlighting
+ */
+const ExampleFooter = ({ html, highlightedJsx, sourceLink }: ExampleFooterProps) => {
+  const [snippetState, setSnippetState] = useState<SnippetState>(SnippetState.CLOSED);
+  const highlightedHtml = html && highlightHtmlSyntax(html);
+
+  const snippetIdPrefix = useRef(uniqueId('example-snippet-')).current;
```

Comment:
[question] what is the benefit of using a ref here instead of setting a regular `const snippetIdPrefix = uniqueId('example-snippet-'))`

Reply:
We do that in some other components, and I don't think we can guarantee that it won't change between renders if we don't. `useRef` makes sure we take the value we get on the first call and stick with that one, even if `uniqueId()` returns a different one on subsequent renders.